terraform/charm/large_deployment/variables.tf

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+variable "cluster_name" {
+  type        = string
+  description = "The cluster name of the fleet."
+  default     = "opensearch"
+}
+
+variable "main" {
+  description = "Main orchestrator app definition"
+  type = object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), {})
+    channel           = optional(string, "2/stable")
+    base              = optional(string, "ubuntu@22.04")
+    revision          = optional(string, null)
+    units             = optional(number, 3)
+    constraints       = optional(string, "arch=amd64")
+    machines          = optional(list(string), [])
+    storage           = optional(map(string), {})
+    endpoint_bindings = optional(map(string), {})
+  })
+}
+
+variable "failover" {
+  description = "Failover orchestrator app definition"
+  type = object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), { "init_hold" : "true" })
```

Comment:
Why is `init_hold` set as default here? Shouldn't that managed by the module internally and never exposed? Why not setting a `cluster_name` default as we need it for the large deployments as well?

Reply:
`init_hold` can never be **not** exposed, as it is a config option, the user can just fill all the config options they wish. 
This ensures that we always override the user input configs for crucial bits, such as `init_hold`

Reply:
Ah, I see the `merge` there. Okay.